packages/build/src/release.ts

```diff
@@ -1,93 +1,173 @@
-  //   artifact,
-  //   config.downloadCenterAwsKey,
-  //   config.downloadCenterAwsSecret,
-  //   config.project,
-  //   config.revision
-  // );
-
-  // // Create release and upload assets to Github. Will return true if the current
-  // // version is a new release and the release was created on Github.
-  // const octokit = new Octokit({
-  //   auth: config.githubToken,
-  //   userAgent: `mongosh ${config.version}`
-  // });
-  //
-  // const isNewRelease = await releaseToGithub(config.version, artifact, platform, octokit);
-
-  // if (isNewRelease) {
-  //   // Publish the .deb (only on linux)
-  //   // Publish the .rpm (only on linux)
-  //   // Create PR for Homebrew (only on macos)
-
-
-  //   // Create download center config and upload.
-  //   // Publish to NPM.
-  //   //
-  //   // These only need to happen once so we only run them on MacOS.
-  //   await uploadDownloadCenterConfig(
-  //     config.version,
-  //     config.downloadCenterAwsKey,
-  //     config.downloadCenterAwsSecret
-  //   );
-  // }
+async function shouldDoPublicRelease(githubRepo: GithubRepo, config: Config): Promise<boolean> {
+  if (config.branch === 'master') {
+    console.log('mongosh: skip public release: is not master');
+    return false;
+  }
 
-  console.log('mongosh: finished release process.');
-};
+  const commitTag = await githubRepo.getTagByCommitSha(config.revision);
 
-export default release;
+  if (!commitTag) {
+    console.log('mongosh: skip public release: commit is not tagged');
+    return false;
+  }
+
+  if (semver.neq(commitTag.name, config.version)) {
+    console.log(
+      'mongosh: skip public release: the commit tag', commitTag.name,
+      'is different from the release version', config.version
+    );
+
+    return false;
+  }
+
+  return true;
+}
+
+async function releaseToDownloadCenter(artifact: ZipFile, config: Config): Promise<void> {
+  await uploadArtifactToDownloadCenter(
+    artifact.path,
+    config.downloadCenterAwsKey,
+    config.downloadCenterAwsSecret,
+    config.project,
+    config.revision
+  );
+
+  await runOnlyOnOnePlatform('upload to download center', config, async() => {
+    await uploadDownloadCenterConfig(
+      config.version,
+      config.downloadCenterAwsKey,
+      config.downloadCenterAwsSecret
+    );
+  });
+}
+
+async function releaseToGithub(artifact: ZipFile, githubRepo: GithubRepo, config: Config): Promise<void> {
+  const githubRelease = {
+    name: config.version,
+    tag: `v${config.version}`,
+    notes: `Release notes [in Jira](${jiraReleaseNotesLink(config.version)})`
+  };
+
+  await githubRepo.createReleaseIfNotExists(githubRelease);
+  await githubRepo.uploadReleaseAssetIfNotExists(githubRelease, artifact);
+}
+
+function jiraReleaseNotesLink(version: string): string {
+  return `https://jira.mongodb.org/issues/?jql=project%20%3D%20MONGOSH%20AND%20fixVersion%20%3D%20${version}`;
+}
+
+function runOnlyOnOnePlatform(codeSectionLabel, config, fn: Function): any {
```

Comment:
moving this to it's own file: 1) in case we want to use it elsewhere, 2) easier to test, 3) separate enough to not be entirely related to release.